### PR TITLE
Add team scale command & add admin chat feedback for other commands

### PR DESCRIPTION
**Pull request by cswhite2000**
Adds a team / ffa scale command to proportionally scale team sizes to their current sizes.
- `/team scale <team> <factor>`
- `/ffa scale <factor>`

Also moves command feedback for certain commands to admin chat.
These are:
- Team management commands
- `/cancel`
- `/start`
- `/timelimit`

### CODE REVIEW

File: core/src/main/java/tc/oc/pgm/command/CancelCommand.java [hunk 1]
```diff
@@ -35,6 +39,9 @@ public void cancel(Audience audience, Match match) {
 
     match.getCountdown().cancelAll();
     match.needModule(StartMatchModule.class).setAutoStart(false);
-    audience.sendMessage(translatable("admin.cancelCountdowns", NamedTextColor.GREEN));
+    ChatDispatcher.broadcastAdminChatMessage(
+        translatable(
+            "admin.cancelCountdowns.announce", PlayerComponent.player(sender, NameStyle.FANCY)),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/CancelCommand.java]:
```suggestion
            "admin.cancelCountdowns.announce", player(sender, NameStyle.FANCY)),
```
We use static imports for components

File: core/src/main/java/tc/oc/pgm/command/StartCommand.java [hunk 2]
```diff
@@ -41,5 +48,11 @@ public void start(
 
     match.getCountdown().cancelAll(StartCountdown.class);
     start.forceStartCountdown(duration, null);
+    ChatDispatcher.broadcastAdminChatMessage(
+        translatable(
+            "admin.start.announce",
+            PlayerComponent.player(sender, NameStyle.FANCY),
+            TemporalComponent.duration(duration, NamedTextColor.AQUA)),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/StartCommand.java]: same here

File: core/src/main/java/tc/oc/pgm/command/TeamCommand.java [hunk 3]
```diff
@@ -49,21 +51,25 @@ public void force(
...
       join.forceJoin(joiner, (Competitor) team);
     }
-
-    sender.sendMessage(
+    ChatDispatcher.broadcastAdminChatMessage(
         translatable(
-            "join.ok.force",
-            NamedTextColor.GRAY,
+            "join.ok.force.announce",
+            PlayerComponent.player(sender, NameStyle.FANCY),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/TeamCommand.java]: and here

File: core/src/main/java/tc/oc/pgm/command/TeamCommand.java [hunk 4]
```diff
@@ -74,14 +80,17 @@ public void shuffle(
       teams.forceJoin(player, null);
     }
 
-    match.sendMessage(translatable("match.shuffle.ok", NamedTextColor.GREEN));
+    ChatDispatcher.broadcastAdminChatMessage(
+        translatable("match.shuffle.announce.ok", PlayerComponent.player(sender, NameStyle.FANCY)),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/TeamCommand.java]: and here

File: core/src/main/java/tc/oc/pgm/command/TeamCommand.java [hunk 5]
```diff
@@ -98,14 +107,44 @@ public void alias(
     final Component oldName = team.getName().color(NamedTextColor.GRAY);
     team.setName(name);
 
-    match.sendMessage(translatable("match.alias.ok", oldName, team.getName()));
+    ChatDispatcher.broadcastAdminChatMessage(
+        translatable(
+            "match.alias.announce.ok",
+            PlayerComponent.player(sender, NameStyle.FANCY),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/TeamCommand.java]: and here

File: core/src/main/java/tc/oc/pgm/command/TeamCommand.java [hunk 6]
```diff
@@ -98,14 +107,44 @@ public void alias(
...
+      @Argument("factor") double scale) {
+    for (Team team : teams) {
+      int maxOverfill = (int) (team.getMaxOverfill() * scale);
+      int maxSize = (int) (team.getMaxPlayers() * scale);
+      team.setMaxSize(maxSize, maxOverfill);
+
+      ChatDispatcher.broadcastAdminChatMessage(
+          translatable(
+              "match.resize.announce.max",
+              PlayerComponent.player(sender, NameStyle.FANCY),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/TeamCommand.java]: and here

File: core/src/main/java/tc/oc/pgm/command/TeamCommand.java [hunk 7]
```diff
@@ -116,49 +155,66 @@ public void max(
       else TextParser.assertInRange(maxOverfill, Range.atLeast(maxPlayers));
 
       team.setMaxSize(maxPlayers, maxOverfill);
-      audience.sendMessage(
+      ChatDispatcher.broadcastAdminChatMessage(
           translatable(
-              "match.resize.max", team.getName(), text(team.getMaxPlayers(), NamedTextColor.AQUA)));
+              "match.resize.announce.max",
+              PlayerComponent.player(sender, NameStyle.FANCY),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/TeamCommand.java]: and here

File: core/src/main/java/tc/oc/pgm/command/TeamCommand.java [hunk 8]
```diff
@@ -116,49 +155,66 @@ public void max(
...
-  public void max(Audience audience, @Argument("teams") Collection<Team> teams) {
+  public void max(CommandSender sender, Match match, @Argument("teams") Collection<Team> teams) {
     for (Team team : teams) {
       team.resetMaxSize();
-      audience.sendMessage(
+      ChatDispatcher.broadcastAdminChatMessage(
           translatable(
-              "match.resize.max", team.getName(), text(team.getMaxPlayers(), NamedTextColor.AQUA)));
+              "match.resize.announce.max",
+              PlayerComponent.player(sender, NameStyle.FANCY),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/TeamCommand.java]: and here

File: core/src/main/java/tc/oc/pgm/command/TeamCommand.java [hunk 9]
```diff
@@ -116,49 +155,66 @@ public void max(
...
       @Argument("min-players") int minPlayers) {
     TextParser.assertInRange(minPlayers, Range.atLeast(0));
     for (Team team : teams) {
       team.setMinSize(minPlayers);
-      audience.sendMessage(
+      ChatDispatcher.broadcastAdminChatMessage(
           translatable(
-              "match.resize.min", team.getName(), text(team.getMinPlayers(), NamedTextColor.AQUA)));
+              "match.resize.announce.min",
+              PlayerComponent.player(sender, NameStyle.FANCY),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/TeamCommand.java]: and here

File: core/src/main/java/tc/oc/pgm/command/TeamCommand.java [hunk 10]
```diff
@@ -116,49 +155,66 @@ public void max(
...
-  public void min(Audience audience, @Argument("teams") Collection<Team> teams) {
+  public void min(CommandSender sender, Match match, @Argument("teams") Collection<Team> teams) {
     for (Team team : teams) {
       team.resetMinSize();
-      audience.sendMessage(
+      ChatDispatcher.broadcastAdminChatMessage(
           translatable(
-              "match.resize.min", team.getName(), text(team.getMinPlayers(), NamedTextColor.AQUA)));
+              "match.resize.announce.min",
+              PlayerComponent.player(sender, NameStyle.FANCY),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/TeamCommand.java]: and here

File: core/src/main/java/tc/oc/pgm/command/TimeLimitCommand.java [hunk 11]
```diff
@@ -45,11 +48,12 @@ public void timelimit(
             true));
     time.start();
 
-    audience.sendMessage(
+    ChatDispatcher.broadcastAdminChatMessage(
         translatable(
-            "match.timeLimit.commandOutput",
-            NamedTextColor.YELLOW,
+            "match.timeLimit.announce.commandOutput",
+            PlayerComponent.player(sender, NameStyle.FANCY),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/TimeLimitCommand.java]: and here

**Review by Pablete1234 — CHANGES_REQUESTED** [review]

**Review by Pablete1234 — APPROVED** [review]

**Review by Electroid — APPROVED** [review]

**Pull request merged**